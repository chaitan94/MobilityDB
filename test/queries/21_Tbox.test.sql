/*****************************************************************************
 * Tbox
 *****************************************************************************/

SELECT tbox 'STBOX((1.0, 2.0), (1.0, 2.0))';

/* Errors */
SELECT tbox '(1, 2)';
SELECT tbox '((AA, 2))';
SELECT tbox '((1, AA))';
SELECT tbox '((1, 2, 3))';
SELECT tbox '((1, 2),(AA, 4))';
SELECT tbox '((1, 2),(3, AA))';
SELECT tbox '((1, 2),(3, 4)';

SELECT tbox(1,2,3,4);
SELECT tbox(4,3,2,1);

SELECT tbox_cmp(tbox '((1.0, 2.0), (1.0, 2.0))', tbox '((2.0, 2.0), (1.0, 2.0))'); 
SELECT tbox_cmp(tbox '((2.0, 2.0), (1.0, 2.0))', tbox '((1.0, 2.0), (1.0, 2.0))'); 
SELECT tbox_cmp(tbox '((1.0, 2.0), (1.0, 2.0))', tbox '((1.0, 3.0), (1.0, 2.0))'); 
SELECT tbox_cmp(tbox '((1.0, 3.0), (1.0, 2.0))', tbox '((1.0, 2.0), (1.0, 2.0))'); 
SELECT tbox_cmp(tbox '((1.0, 2.0), (1.0, 2.0))', tbox '((1.0, 2.0), (2.0, 2.0))'); 
SELECT tbox_cmp(tbox '((1.0, 2.0), (2.0, 2.0))', tbox '((1.0, 2.0), (1.0, 2.0))'); 
SELECT tbox_cmp(tbox '((1.0, 2.0), (1.0, 2.0))', tbox '((1.0, 2.0), (1.0, 3.0))'); 
SELECT tbox_cmp(tbox '((1.0, 2.0), (1.0, 3.0))', tbox '((1.0, 2.0), (1.0, 2.0))'); 

SELECT tbox_cmp(tbox '((1.0, 2.0), (1.0, 2.0))', tbox '((1.0, 2.0), (1.0, 2.0))'); 

SELECT tbox '((1.0, 2.0), (1.0, 2.0))' = floatrange '[1, 2]'::tbox; 

-------------------------------------------------------------------------------

SELECT count(*) FROM tbl_tbox t1, tbl_tbox t2 where t1.b = t2.b;
SELECT count(*) FROM tbl_tbox t1, tbl_tbox t2 where t1.b <> t2.b;
SELECT count(*) FROM tbl_tbox t1, tbl_tbox t2 where t1.b < t2.b;
SELECT count(*) FROM tbl_tbox t1, tbl_tbox t2 where t1.b <= t2.b;
SELECT count(*) FROM tbl_tbox t1, tbl_tbox t2 where t1.b > t2.b;
SELECT count(*) FROM tbl_tbox t1, tbl_tbox t2 where t1.b >= t2.b;

SELECT count(*) FROM tbl_tfloat WHERE temp::tbox IS NOT NULL;
SELECT count(*) FROM tbl_tfloat WHERE temp::tbox IS NOT NULL;

SELECT count(*) FROM tbl_tbox t1, tbl_tbox t2 where t1.b && t2.b;
SELECT count(*) FROM tbl_tbox t1, tbl_tbox t2 where t1.b @> t2.b;
SELECT count(*) FROM tbl_tbox t1, tbl_tbox t2 where t1.b <@ t2.b;
SELECT count(*) FROM tbl_tbox t1, tbl_tbox t2 where t1.b ~= t2.b;

/*****************************************************************************/