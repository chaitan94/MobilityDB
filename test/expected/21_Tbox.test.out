/*****************************************************************************
 * Tbox
 *****************************************************************************/
SELECT tbox 'TBOX((1.0, 2.0), (1.0, 2.0))';
       tbox        
-------------------
 TBOX((1,2),(1,2))
(1 row)

SELECT tbox 'TBOX((1.0,), (1.0,))';
      tbox       
-----------------
 TBOX((1,),(1,))
(1 row)

SELECT tbox 'TBOX((, 2.0), (, 2.0))';
      tbox       
-----------------
 TBOX((,2),(,2))
(1 row)

SELECT tbox 'TBOX((4,3),(2,1))';
       tbox        
-------------------
 TBOX((2,1),(4,3))
(1 row)

/* Errors */
SELECT tbox 'STBOX(1, 2)';
ERROR:  Could not parse TBOX
LINE 2: SELECT tbox 'STBOX(1, 2)';
                    ^
SELECT tbox 'TBOX(1, 2)';
ERROR:  Could not parse TBOX
LINE 1: SELECT tbox 'TBOX(1, 2)';
                    ^
SELECT tbox 'TBOX((AA, 2))';
ERROR:  Could not parse TBOX
LINE 1: SELECT tbox 'TBOX((AA, 2))';
                    ^
SELECT tbox 'TBOX((1, AA))';
ERROR:  Could not parse TBOX
LINE 1: SELECT tbox 'TBOX((1, AA))';
                    ^
SELECT tbox 'TBOX((1, 2, 3))';
ERROR:  Could not parse TBOX
LINE 1: SELECT tbox 'TBOX((1, 2, 3))';
                    ^
SELECT tbox 'TBOX((1, 2),3, 4))';
ERROR:  Could not parse TBOX
LINE 1: SELECT tbox 'TBOX((1, 2),3, 4))';
                    ^
SELECT tbox 'TBOX((1, 2),(AA, 4))';
ERROR:  Could not parse STBOX
LINE 1: SELECT tbox 'TBOX((1, 2),(AA, 4))';
                    ^
SELECT tbox 'TBOX((1, 2),(3, AA))';
ERROR:  Could not parse STBOX
LINE 1: SELECT tbox 'TBOX((1, 2),(3, AA))';
                    ^
SELECT tbox 'TBOX((1, 2),(3, 4)';
ERROR:  Could not parse TBOX
LINE 1: SELECT tbox 'TBOX((1, 2),(3, 4)';
                    ^
SELECT tbox(1,2,3,4);
       tbox        
-------------------
 TBOX((1,2),(3,4))
(1 row)

SELECT tbox(4,3,2,1);
       tbox        
-------------------
 TBOX((2,1),(4,3))
(1 row)

SELECT tbox(1,2);
      tbox       
-----------------
 TBOX((1,),(2,))
(1 row)

SELECT tbox(2,1);
      tbox       
-----------------
 TBOX((1,),(2,))
(1 row)

SELECT tboxt(1,2);
      tboxt      
-----------------
 TBOX((,1),(,2))
(1 row)

SELECT tboxt(2,1);
      tboxt      
-----------------
 TBOX((,1),(,2))
(1 row)

SELECT tbox_cmp(tbox 'TBOX((1.0, 2.0), (1.0, 2.0))', tbox 'TBOX((2.0, 2.0), (1.0, 2.0))');
 tbox_cmp 
----------
       -1
(1 row)

SELECT tbox_cmp(tbox 'TBOX((2.0, 2.0), (1.0, 2.0))', tbox 'TBOX((1.0, 2.0), (1.0, 2.0))');
 tbox_cmp 
----------
        1
(1 row)

SELECT tbox_cmp(tbox 'TBOX((1.0, 2.0), (1.0, 2.0))', tbox 'TBOX((1.0, 3.0), (1.0, 2.0))');
 tbox_cmp 
----------
       -1
(1 row)

SELECT tbox_cmp(tbox 'TBOX((1.0, 3.0), (1.0, 2.0))', tbox 'TBOX((1.0, 2.0), (1.0, 2.0))');
 tbox_cmp 
----------
        1
(1 row)

SELECT tbox_cmp(tbox 'TBOX((1.0, 2.0), (1.0, 2.0))', tbox 'TBOX((1.0, 2.0), (2.0, 2.0))');
 tbox_cmp 
----------
       -1
(1 row)

SELECT tbox_cmp(tbox 'TBOX((1.0, 2.0), (2.0, 2.0))', tbox 'TBOX((1.0, 2.0), (1.0, 2.0))');
 tbox_cmp 
----------
        1
(1 row)

SELECT tbox_cmp(tbox 'TBOX((1.0, 2.0), (1.0, 2.0))', tbox 'TBOX((1.0, 2.0), (1.0, 3.0))');
 tbox_cmp 
----------
       -1
(1 row)

SELECT tbox_cmp(tbox 'TBOX((1.0, 2.0), (1.0, 3.0))', tbox 'TBOX((1.0, 2.0), (1.0, 2.0))');
 tbox_cmp 
----------
        1
(1 row)

SELECT tbox_cmp(tbox 'TBOX((1.0, 2.0), (1.0, 2.0))', tbox 'TBOX((1.0, 2.0), (1.0, 2.0))');
 tbox_cmp 
----------
        0
(1 row)

SELECT tbox 'TBOX((1.0, 2.0), (1.0, 2.0))' = floatrange '[1, 2]'::tbox;
 ?column? 
----------
 f
(1 row)

SELECT count(*) FROM tbl_tbox t1, tbl_tbox t2 where t1.b = t2.b;
 count 
-------
    99
(1 row)

SELECT count(*) FROM tbl_tbox t1, tbl_tbox t2 where t1.b <> t2.b;
 count 
-------
  9702
(1 row)

SELECT count(*) FROM tbl_tbox t1, tbl_tbox t2 where t1.b < t2.b;
 count 
-------
  4851
(1 row)

SELECT count(*) FROM tbl_tbox t1, tbl_tbox t2 where t1.b <= t2.b;
 count 
-------
  4950
(1 row)

SELECT count(*) FROM tbl_tbox t1, tbl_tbox t2 where t1.b > t2.b;
 count 
-------
  4851
(1 row)

SELECT count(*) FROM tbl_tbox t1, tbl_tbox t2 where t1.b >= t2.b;
 count 
-------
  4950
(1 row)

SELECT count(*) FROM tbl_tfloat WHERE temp::tbox IS NOT NULL;
 count 
-------
    96
(1 row)

SELECT count(*) FROM tbl_tfloat WHERE temp::tbox IS NOT NULL;
 count 
-------
    96
(1 row)

SELECT count(*) FROM tbl_tbox t1, tbl_tbox t2 where t1.b && t2.b;
 count 
-------
   215
(1 row)

SELECT count(*) FROM tbl_tbox t1, tbl_tbox t2 where t1.b @> t2.b;
 count 
-------
   107
(1 row)

SELECT count(*) FROM tbl_tbox t1, tbl_tbox t2 where t1.b <@ t2.b;
 count 
-------
   107
(1 row)

SELECT count(*) FROM tbl_tbox t1, tbl_tbox t2 where t1.b ~= t2.b;
 count 
-------
    99
(1 row)

/*****************************************************************************/
